src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -0,0 +1,349 @@
+package scala                     //TODO: Change this strawman --> we just need sizeHintIfCheap from scala
```

Comment:
IMO, `sizeIfCheap` makes more sense here then `knownSize` since former implies size is not known because of expensive-ness and the latter implies size is not known because of something could be an infinite stream